### PR TITLE
GAWB-4055: Fix visibility of datasets that have a discoverable by group set

### PR DESCRIPTION
**Pull request by ahaessly**
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by dvoet — APPROVED** [review]

File: src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryServiceSupport.scala [hunk 1]
```diff
@@ -168,7 +167,11 @@ trait LibraryServiceSupport extends DataUseRestrictionSupport with LazyLogging {
   }
 
   def getEffectiveDiscoverGroups(samDAO: SamDAO)(implicit ec: ExecutionContext, userInfo:UserInfo): Future[Seq[String]] = {
-    samDAO.listGroups(userInfo) map {FireCloudConfig.ElasticSearch.discoverGroupNames intersect _}
+    samDAO.listGroups(userInfo) map { groupMemberships =>
+      groupMemberships map { groupMembership =>
+        groupMembership.groupName
+      } intersect FireCloudConfig.ElasticSearch.discoverGroupNames
+    }
```
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryServiceSupport.scala:172]: you could make this a bit shorter - your call on code style if you want to implement this way or not:
```
    samDAO.listGroups(userInfo) map { groupMemberships =>
      groupMemberships map (_.groupName) intersect FireCloudConfig.ElasticSearch.discoverGroupNames
    }
```

**Review by davidangb — APPROVED** [review]
:+1: one non-blocking suggestion inline.

File: src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryServiceSupport.scala (hunk 1 above)
Comment (ahaessly) [reply src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryServiceSupport.scala:172]: i tried several different ways to simplify - i knew you would have the winning one!

**Review by davidangb — APPROVED** [review]
still 👍

**Pull request merged**